### PR TITLE
ERMS-3072: Umfrage-Bugs in 2.0 

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: grails-app/controllers/de/laser/SurveyController.groovy [hunk 1]
```diff
@@ -345,10 +348,10 @@ class SurveyController {
         result.subscriptions = subscriptions.drop((int) result.offset).take((int) result.max)
 
         result.allLinkedLicenses = [:]
-        Set<Links> allLinkedLicenses = Links.findAllByDestinationInListAndLinkType(result.subscriptions.collect { Subscription s -> genericOIDService.getOID(s) },RDStore.LINKTYPE_LICENSE)
+        Set<Links> allLinkedLicenses = Links.findAllByDestinationSubscriptionInListAndLinkType(result.subscriptions ,RDStore.LINKTYPE_LICENSE)
         allLinkedLicenses.each { Links li ->
-            Subscription s = (Subscription) genericOIDService.resolveOID(li.destination)
-            License l = (License) genericOIDService.resolveOID(li.source)
+            Subscription s = (Subscription) genericOIDService.resolveOID(li.destinationSubscription)
```
Comment (croqueGrec09) [inline grails-app/controllers/de/laser/SurveyController.groovy:353]: li.destinationSubscription does not need to be converted, is already a subscription and not an OID any more

File: grails-app/domain/de/laser/Subscription.groovy [hunk 2]
```diff
@@ -377,8 +377,8 @@ class Subscription extends AbstractBaseWithCalculatedLastUpdated
 
     Set<License> getLicenses() {
         Set<License> result = []
-        Links.findAllByDestinationAndLinkType(genericOIDService.getOID(this),RDStore.LINKTYPE_LICENSE).each { l ->
-            result << genericOIDService.resolveOID(l.source)
+        Links.findAllByDestinationSubscriptionAndLinkType(genericOIDService.getOID(this),RDStore.LINKTYPE_LICENSE).each { l ->
+            result << genericOIDService.resolveOID(l.sourceLicense)
```
Comment (croqueGrec09) [inline grails-app/domain/de/laser/Subscription.groovy:381]: see above

File: grails-app/services/de/laser/CopyElementsService.groovy [hunk 3]
```diff
@@ -229,7 +229,7 @@ class CopyElementsService {
 //                diffs.add(message(code:'pendingChange.message_CI01',args:[costTitle,g.createLink(mapping:'subfinance',controller:'subscription',action:'index',params:[sub:cci.sub.id]),cci.sub.name,cci.costInBillingCurrency,newCostItem
             } else {
                 //ChildSub Exist
-//                ArrayList<Links> prevLinks = Links.findAllByDestinationAndLinkTypeAndObjectType(subMember.id, RDStore.LINKTYPE_FOLLOWS, Subscription.class.name)
+//                ArrayList<Links> prevLinks = Links.findAllByDestinationLicenseAndLinkTypeAndObjectType(subMember.id, RDStore.LINKTYPE_FOLLOWS, Subscription.class.name)
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/CopyElementsService.groovy:232]: ObjectType is not existing any more, that is why DestinationLicense (rather DestinationSubscription here) would have to be queried - if the section had not been commented out anyway

File: grails-app/services/de/laser/LinksGenerationService.groovy [hunk 4]
```diff
@@ -108,13 +108,13 @@ class LinksGenerationService {
             //IN is from the point of view of the context object (= obj)
 
             sources.each { Links link ->
-                Subscription destination = (Subscription) genericOIDService.resolveOID(link.destination)
+                Subscription destination = (Subscription) genericOIDService.resolveOID(link.destinationSubscription)
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/LinksGenerationService.groovy:111]: see above

File: grails-app/services/de/laser/SurveyService.groovy [hunk 5]
```diff
@@ -270,8 +271,8 @@ class SurveyService {
                         row.add([field: subscription?.agencies ? subscription?.agencies?.join(", ") : '', style: null])
 
                         List licenseNames = []
-                        Links.findAllByDestinationAndLinkType(genericOIDService.getOID(subscription),RDStore.LINKTYPE_LICENSE).each { Links li ->
-                            License l = (License) genericOIDService.resolveOID(li.source)
+                        Links.findAllByDestinationSubscriptionAndLinkType(genericOIDService.getOID(subscription),RDStore.LINKTYPE_LICENSE).each { Links li ->
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SurveyService.groovy:274]: see above

File: grails-app/services/de/laser/SurveyService.groovy [hunk 6]
```diff
@@ -734,8 +735,8 @@ class SurveyService {
                             row.add([field: subscription?.agencies ? subscription?.agencies?.join(", ") : '', style: null])
 
                             List licenseNames = []
-                            Links.findAllByDestinationAndLinkType(genericOIDService.getOID(subscription),RDStore.LINKTYPE_LICENSE).each { Links li ->
-                                License l = (License) genericOIDService.resolveOID(li.source)
+                            Links.findAllByDestinationSubscriptionAndLinkType(genericOIDService.getOID(subscription),RDStore.LINKTYPE_LICENSE).each { Links li ->
+                                License l = (License) genericOIDService.resolveOID(li.sourceLicense)
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SurveyService.groovy:739]: see above

**Review by croqueGrec09 — CHANGES_REQUESTED** [review]
Links retrieval has a new structure

File: grails-app/controllers/de/laser/SurveyController.groovy [hunk 7]
```diff
@@ -2786,7 +2794,7 @@ class SurveyController {
             result.orgsWithoutResult.each { surveyResult ->
                 if (surveyResult.participant.id in currentParticipantIDs && surveyResult.sub) {
 
-                    if (property.type == RefdataValue.CLASS) {
+                    if (property.type == RefdataValue.class.name) {
```
Comment (klodav) [inline grails-app/controllers/de/laser/SurveyController.groovy:2797]: please use PropertyDefinition.isRefdataValueType()

File: grails-app/controllers/de/laser/YodaController.groovy [hunk 8]
```diff
@@ -1777,7 +1777,7 @@ class YodaController {
                     }
 
                 } else {
-                   /* if (property?.type == RefdataValue.CLASS) {
+                   /* if (property?.type == RefdataValue.class.name) {
```
Comment (klodav) [inline grails-app/controllers/de/laser/YodaController.groovy:1780]: please use PropertyDefinition.isRefdataValueType()

File: grails-app/domain/de/laser/SurveyProperty.groovy [hunk 9]
```diff
@@ -117,7 +117,7 @@ class SurveyProperty implements I10nTrait {
 
         def propertyType = SurveyProperty.getLocalizedValue(this.type)
         List refdataValues = []
-        if(this.type == RefdataValue.CLASS){
+        if(this.type == RefdataValue.class.name){
```
Comment (klodav) [inline grails-app/domain/de/laser/SurveyProperty.groovy:120]: please use PropertyDefinition.isRefdataValueType()

File: grails-app/views/profile/properties.gsp [hunk 10]
```diff
@@ -44,7 +44,7 @@
                             <td>${pd.getI10n('expl', 'en')}</td>
                             <td>
                                 <g:set var="pdRdc" value="${pd.type?.split('\\.').last()}"/>
-                                <g:if test="${pd?.type == RefdataValue.CLASS}">
+                                <g:if test="${pd?.type == RefdataValue.class.name}">
```
Comment (klodav) [inline grails-app/views/profile/properties.gsp:47]: please use PropertyDefinition.isRefdataValueType()

File: grails-app/views/survey/allSurveyProperties.gsp [hunk 11]
```diff
@@ -96,7 +96,7 @@
                 </td>
                 <td>
                     ${PropertyDefinition.getLocalizedValue(property.type)}
-                    <g:if test="${property.type == RefdataValue.CLASS}">
+                    <g:if test="${property.type == RefdataValue.class.name}">
```
Comment (klodav) [inline grails-app/views/survey/allSurveyProperties.gsp:99]: please use PropertyDefinition.isRefdataValueType()

File: grails-app/views/templates/copyElements/_propertyComparisonTableRow.gsp [hunk 12]
```diff
@@ -203,7 +203,7 @@
                             <div class="la-copyElements-flex-container la-multi-sources la-colorCode-source">
                                 <div class="la-copyElements-flex-item">
                                     ${PropertyDefinition.getLocalizedValue(propValue.type)}
-                                    <g:if test="${propValue.type == RefdataValue.CLASS}">
+                                    <g:if test="${propValue.type == RefdataValue.class.name}">
```
Comment (klodav) [inline grails-app/views/templates/copyElements/_propertyComparisonTableRow.gsp:206]: please use PropertyDefinition.isRefdataValueType()

File: grails-app/views/templates/survey/_generalSurvey.gsp [hunk 13]
```diff
@@ -293,7 +293,7 @@
                     <td>
 
                         ${PropertyDefinition.getLocalizedValue(surveyProperty.surveyProperty.type)}
-                        <g:if test="${surveyProperty.surveyProperty.type == RefdataValue.CLASS}">
+                        <g:if test="${surveyProperty.surveyProperty.type == RefdataValue.class.name}">
```
Comment (klodav) [inline grails-app/views/templates/survey/_generalSurvey.gsp:296]: please use PropertyDefinition.isRefdataValueType()

File: grails-app/views/templates/survey/_subscriptionSurvey.gsp [hunk 14]
```diff
@@ -890,7 +890,7 @@
                     </td>
                     <td>
                         ${PropertyDefinition.getLocalizedValue(surveyProperty.surveyProperty.type)}
-                        <g:if test="${surveyProperty.surveyProperty.type == RefdataValue.CLASS}">
+                        <g:if test="${surveyProperty.surveyProperty.type == RefdataValue.class.name}">
```
Comment (klodav) [inline grails-app/views/templates/survey/_subscriptionSurvey.gsp:893]: please use PropertyDefinition.isRefdataValueType()

**Review by croqueGrec09 — APPROVED** [review]

**Pull request merged**